xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -327,485 +366,612 @@ XdsClient createXdsClient() {
-
-    lookasideLb.shutdown();
+    fakeClock.forwardNanos(TimeUnit.SECONDS.toNanos(20));
+    verify(edsUpdateCallback).onError();
   }
 
   @Test
-  public void verifyRpcErrorPropagation() {
-    lookasideLb.handleResolvedAddresses(defaultResolvedAddress);
+  public void verifyErrorPropagation_withXdsClientPoolAttributes() {
+    setUpWithXdsClientPoolAttributes();
+    handleResolvedAddresses(new XdsConfig(null, null, "edsServiceName1", null));
 
     verify(edsUpdateCallback, never()).onError();
-    serverResponseWriter.onError(new RuntimeException());
+    endpointWatchers.get("edsServiceName1")
+        .onError(Status.DATA_LOSS.withDescription("fake_status"));
     verify(edsUpdateCallback).onError();
   }
 
-  @Test
-  public void shutdown() {
-    lookasideLb.handleResolvedAddresses(defaultResolvedAddress);
+  /**
+   * Converts ClusterLoadAssignment data to {@link EndpointUpdate}. All the needed data, that is
+   * clusterName, localityLbEndpointsMap and dropPolicies, is extracted from ClusterLoadAssignment,
+   * and all other data is ignored.
+   */
+  private static EndpointUpdate getEndpointUpdateFromClusterAssignment(
+      ClusterLoadAssignment clusterLoadAssignment) {
+    EndpointUpdate.Builder endpointUpdateBuilder = EndpointUpdate.newBuilder();
+    endpointUpdateBuilder.setClusterName(clusterLoadAssignment.getClusterName());
+    for (DropOverload dropOverload : clusterLoadAssignment.getPolicy().getDropOverloadsList()) {
+      endpointUpdateBuilder.addDropPolicy(
+          EnvoyProtoData.DropOverload.fromEnvoyProtoDropOverload(dropOverload));
+    }
+    for (LocalityLbEndpoints localityLbEndpoints : clusterLoadAssignment.getEndpointsList()) {
+      endpointUpdateBuilder.addLocalityLbEndpoints(
+          EnvoyProtoData.Locality.fromEnvoyProtoLocality(
+              localityLbEndpoints.getLocality()),
+          EnvoyProtoData.LocalityLbEndpoints.fromEnvoyProtoLocalityLbEndpoints(
+              localityLbEndpoints));
+    }
+    return endpointUpdateBuilder.build();
+  }
 
-    LocalityStore localityStore = Iterables.getOnlyElement(localityStores);
-    LoadReportClient loadReportClient = Iterables.getOnlyElement(loadReportClients);
-    verify(localityStore, never()).reset();
-    verify(loadReportClient, never()).stopLoadReporting();
-    assertThat(channel.isShutdown()).isFalse();
+  private void handleResolvedAddresses(XdsConfig xdsConfig) {
+    ResolvedAddresses.Builder resolvedAddressBuilder = ResolvedAddresses.newBuilder()
+        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
+        .setLoadBalancingPolicyConfig(xdsConfig);
+    if (isFullFlow) {
+      resolvedAddressBuilder.setAttributes(
+          Attributes.newBuilder().set(XdsAttributes.XDS_CLIENT_POOL,
+              xdsClientPoolFromResolveAddresses).build());
+    }
+    lookasideLb.handleResolvedAddresses(resolvedAddressBuilder.build());
+  }
 
-    lookasideLb.shutdown();
+  private int versionIno;
```

Comment:
I purposefully put it right above the only method using it, so that it's easy to read the method. It's a minor style preference. I don't mind to change it.